[Flex] getComputedStyle should reflect that first-letter and first-line cannot be applied to flexboxes.

sammygill opened the pull request:
#### 02a339f40c687d1ce5b504eea903ab94cf8b40ca
<pre>
[Flex] getComputedStyle should reflect that first-letter and first-line cannot be applied to flexboxes.
<a href="https://bugs.webkit.org/show_bug.cgi?id=210474">https://bugs.webkit.org/show_bug.cgi?id=210474</a>
<a href="https://rdar.apple.com/94163778">rdar://94163778</a>

Reviewed by Tim Nguyen.

first-letter and first-line cannot be applied to flexboxes since these
boxes do not generate a first formatted line. However, if you attempt to
query the respective pseudo element&apos;s style via getComputedStyle you
will end up with the value that was specified even though it should not
have been applied.

(WebCore::Document::styleForElementIgnoringPendingStylesheets):
Instead of just trying to resolve the style for the pseudo element
immediately, we should first resolve the style with the associated
element and check whether or not it can have the associated pseudo
content.

Canonical link: <a href="https://commits.webkit.org/298706@main">https://commits.webkit.org/298706@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b02aae86372c30d7303f1fe5e31d043d0a6b757

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116236 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35897 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26443 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122293 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66796 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2af4638a-61ec-43a3-b41f-51e955072631) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118125 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36591 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44485 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88301 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42817 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/67221192-0761-43c3-83d5-c01b0220c09f) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29187 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104288 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68712 "Passed tests") | | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22396 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65974 "Built successfully") | | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22543 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125442 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43130 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32387 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97019 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43495 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100487 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96804 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24658 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42086 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19979 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39077 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43017 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48609 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42484 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45819 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44188 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e93595784c51733ca146751d18fb1b2950cc144d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48454

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/aa7accae0b4b6d4bd3ac9066ddf63905ed25a81d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48454

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-flexbox/getcomputedstyle/first-line-computed-style.html:
```diff
@@ -0,0 +1,19 @@
+
```
nit: extra new line

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/94d3448e82b8df1c7c85fa54110ccabbd3765cf6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48454

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8b02aae86372c30d7303f1fe5e31d043d0a6b757. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48454

sammygill committed:
[Flex] getComputedStyle should reflect that first-letter and first-line cannot be applied to flexboxes.
https://bugs.webkit.org/show_bug.cgi?id=210474
rdar://94163778

Reviewed by Tim Nguyen.

first-letter and first-line cannot be applied to flexboxes since these
boxes do not generate a first formatted line. However, if you attempt to
query the respective pseudo element's style via getComputedStyle you
will end up with the value that was specified even though it should not
have been applied.

(WebCore::Document::styleForElementIgnoringPendingStylesheets):
Instead of just trying to resolve the style for the pseudo element
immediately, we should first resolve the style with the associated
element and check whether or not it can have the associated pseudo
content.

Canonical link: https://commits.webkit.org/298706@main

webkit-commit-queue commented:
Committed 298706@main (02a339f40c68): <https://commits.webkit.org/298706@main>

Reviewed commits have been landed. Closing PR #48454 and removing active labels.
